packages/shorebird_cli/lib/src/shorebird_process.dart

```diff
@@ -26,41 +28,74 @@ abstract class ShorebirdProcess {
     String executable,
     List<String> arguments, {
     bool runInShell = false,
+    Map<String, String>? environment,
     String? workingDirectory,
+    bool resolveExecutables = true,
   }) {
+    final resolvedEnvironment = (environment ?? {})
+      ..addAll(
```

Comment:
You only want to do this if resolveExecutables=true or you'll end up potentially confusing their installed flutter (should be harmless, but no need to confuse things).

Reply:
Ah, yep, good catch. Updated and added tests.